CityPuzzle/Game Room/CreateGamePage.xaml.cs

```diff
@@ -11,40 +13,47 @@ namespace CityPuzzle
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class CreateGamePage : ContentPage
     {
-        public static Room NewRoom;
+        public static Lazy<Room> NewRoom= new Lazy<Room>();
         public List<Lazy<Puzzle>> DefaultPuzzles;
         public List<string> AllUsers;
+        public static Thread Calculiator_thread=new Thread(new ThreadStart(FillGameRomm));
         public static int Status = -1;
-        private readonly Random _random = new Random();
 
         public CreateGamePage()
         {
-            NewRoom = new Room();
-            NewRoom.ID = CreatePin();
-            NewRoom.Owner = App.CurrentUser.ID;
-
-        InitializeComponent();
+            Calculiator_thread.Start();
+            InitializeComponent();
+            addobj.IsVisible = true;
         }
-        
-        public string CreatePin()
+        public async static void FillGameRomm()
         {
-            int roomID = _random.Next(1, 100);
-            string roomPin = "kambarys" + roomID;
-
+            await CreatePin();
+            NewRoom.Value.Owner = App.CurrentUser.ID;        }
+        public async static Task CreatePin()
+        {
+            int i = 0;
             var AllRooms = Sql.ReadRooms();
-            Room existing = AllRooms.SingleOrDefault(x => x.ID.ToLower().Equals(roomPin.ToLower()));
-            if (existing != null) roomPin = CreatePin();
-            return roomPin;
+            string roomPin = "";
+            Random _random = new Random();
+            await Task.Run(() =>
+            {
+                while (i == 0)
+                {
+                    int roomID = _random.Next(100, 10000);
+                    roomPin = "kambarys" + roomID;
+                    Room existing = AllRooms.SingleOrDefault(x => x.ID.ToLower().Equals(roomPin.ToLower()));
+                    if (existing == null) i = 1;
+                }
+            });
+            NewRoom.Value.ID=roomPin;
         }
 
         protected override void OnAppearing()
         {
             base.OnAppearing();
             DefaultPuzzles = Sql.ReadPuzzles();
-            if (Status == -1)
-                {
-                    AddObj_click(null, null);
-                }
+            if (Status == -1)AddObj_click(null, null);
```

Comment:
Please add whitespace after if statement or even better - write code in a new line